modules/playlists/src/main/java/org/opencastproject/playlists/Playlist.java

```diff
@@ -0,0 +1,240 @@
+ * agreements. See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ *
+ * The Apereo Foundation licenses this file to you under the Educational
+ * Community License, Version 2.0 (the "License"); you may not use this file
+ * except in compliance with the License. You may obtain a copy of the License
+ * at:
+ *
+ *   http://opensource.org/licenses/ecl2.txt
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.  See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ *
+ */
+
+package org.opencastproject.playlists;
+
+
+import static org.opencastproject.util.RequireUtil.notNull;
+
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.List;
+import java.util.UUID;
+
+import javax.persistence.CascadeType;
+import javax.persistence.Column;
+import javax.persistence.Entity;
+import javax.persistence.FetchType;
+import javax.persistence.Id;
+import javax.persistence.NamedQueries;
+import javax.persistence.NamedQuery;
+import javax.persistence.OneToMany;
+import javax.persistence.OrderColumn;
+import javax.persistence.PrePersist;
+import javax.persistence.Table;
+import javax.persistence.Temporal;
+import javax.persistence.TemporalType;
+
+/**
+ * Entity object for storing playlists in persistence storage. Playlists contain an ordered list of entries,
+ * which represent Opencast entities (for now they can only hold events, but in the future they may also hold series
+ * and the like). Playlists also contain some metadata and their own access control list.
+ *
+ * The playlist ACL only pertains to itself and its entries. The contents of an entry (i.e. the publication of an
+ * event) is still only governed by the contents ACL.
+ */
+@Entity(name = "Playlist")
+@Table(name = "oc_playlist")
+@NamedQueries({
+    @NamedQuery(
+        name = "Playlist.findById",
+        query = "SELECT p FROM Playlist p WHERE p.id = :id and p.organization = :organizationId"
+    ),
+})
+public class Playlist {
+
+  /**
+   * Workaround for generating a UUID for the id if we don't already have one.
+   *
+   * We cannot use EclipseLinks @UuidGenerator annotation for this, because it breaks our JUnit tests.
+   * We also cannot use Hibernates @GenericGenerator for this, since we lack the appropriate library (and possibly
+   * hibernate version).
+   */
+  @PrePersist
+  public void generateId() {
+    if (this.id == null) {
+      this.id = UUID.randomUUID().toString();
+    }
+  }
+
+  @Id
+  @Column(name = "id")
+  private String id;
+
+  @Column(name = "organization", nullable = false, length = 128)
+  private String organization;
+
+  @OneToMany(
+      fetch = FetchType.LAZY,
+      cascade = { CascadeType.PERSIST, CascadeType.REMOVE },
```

Comment:
The entries do not get updated on merge:
```suggestion
      cascade = { CascadeType.PERSIST, CascadeType.MERGE, CascadeType.REMOVE },
```